tests/reg_tests/test_separation.py

```diff
@@ -0,0 +1,141 @@
+# built-ins
+import unittest
+import numpy as np
+import os
+import copy
+
+# MACH classes
+from adflow import ADFLOW
+
+# import the testing utilities
+from reg_default_options import adflowDefOpts
+from reg_aeroproblems import ap_naca0012_separation
+import reg_test_classes
+
+baseDir = os.path.dirname(os.path.abspath(__file__))
+
+
+class SeparationBasicTests(reg_test_classes.RegTest):
```

Comment:
Could you also add a complex test like in the cavitation tests? Otherwise, this does not test whether the derivatives are accurate.

Reply:
I added a complex tests too, Please check it.